### PR TITLE
Add get event type

### PR DESCRIPTION
**Pull request by ActivePeter**
When implementing waiting for a key to be deleted, the event this lib returned has no func to verify the type of event (Put or delete)

### CODE REVIEW

File: etcd-client/src/client.rs [hunk 1]
```diff
@@ -71,9 +71,8 @@ pub struct Inner {
 impl Client {
     /// Get grpc channel.
     fn get_channel(cfg: &ClientConfig) -> Channel {
-        if cfg.endpoints.is_empty() {
-            panic!("Empty etcd endpoints");
-        }
+        assert!(!cfg.endpoints.is_empty(), "Empty etcd endpoints");
```
Comment (xiaguan) [inline etcd-client/src/client.rs:74]: Why use assert instead of panic ？

File: mock-etcd/Cargo.toml [hunk 2]
```diff
@@ -12,7 +12,7 @@ categories = ["storage"]
 [dependencies]
 async-lock = "2.3.0"
 async-io = "1.6.0"
-env_logger = "0.8.1"
+env_logger = "0.9.3"
```
Comment (xiaguan) [inline mock-etcd/Cargo.toml:15]: Here, are the changes different from the version changes in etcd_client?

File: etcd-client/src/client.rs (hunk 1 above)
Comment (ActivePeter) [reply etcd-client/src/client.rs:74]: Rust lint says it's a better way.

File: mock-etcd/Cargo.toml (hunk 2 above)
Comment (ActivePeter) [reply mock-etcd/Cargo.toml:15]: This change is for compile problem. Old version has some broken dependency relationship.

File: etcd-client/src/lib.rs [hunk 3]
```diff
@@ -77,7 +77,13 @@
...
     clippy::module_name_repetitions, // repeation of module name in a struct name is not big deal
     clippy::multiple_crate_versions, // multi-version dependency crates is not able to fix
-    clippy::panic, // allow debug_assert, panic in production code
+    clippy::panic, // allow debug_assert, panic in production code  
+    clippy::shadow_same, //it’s a common pattern in Rust code
+    clippy::shadow_unrelated,
+    clippy::shadow_reuse,
+    clippy::same_name_method,
+    clippy::separated_literal_suffix,
+    clippy::mod_module_files, // TODO: fix code structure to pass this lint
```
Comment (rogercloud) [inline etcd-client/src/lib.rs]: Why do we allow these clippy?

Comment (ActivePeter) [reply etcd-client/src/lib.rs]: The reasons are added in the recent commit, which are caused by rust version update and there are some conflict clippy rule(we have to choose one of them) and some are for the generated code.

**Review by rogercloud — APPROVED** [review]

**Pull request merged**